### PR TITLE
[INFRA/IIGO] Accountability Monitoring Results Fed Into Power-Transfer

### PR DESCRIPTION
**Pull request by preetl**
# Summary

DO NOT MERGE BEFORE #211 !

Added the monitoring result as an input to the power-transfer functions so that islands can make informed decisions on whether to call an election. 

Part 4 of Accountability Cycle ( 1 more to come with Rules and Integration tests!)

@lhl2617 I've added tests on the function with the questionable switch case that you commented on in my last PR!

## Additional Information

Happy New Year!!! <3

## Test Plan

New tests included and more to come once these features are fully integrated with changes from #208

### CODE REVIEW

**Comment by QFSW** [comment]
# ⚠️Warning! This diff contains changes in protected folders

File: website/README.md [hunk 1]
```diff
@@ -25,9 +25,12 @@ sudo npm i -g yarn
 
 `yarn install`
 
+<<<<<<< HEAD
+=======
 ## Scripts
 
 ### `yarn prestart`
+>>>>>>> main
```
Comment (preetl) [inline website/README.md]:
```suggestion
## Scripts

### `yarn prestart`
```

File: internal/common/baseclient/baseSpeaker.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ func (s *BaseSpeaker) DecideAnnouncement(ruleId string, result bool) (string, bo
 }
 
 // CallJudgeElection is called by the legislature to decide on power-transfer
```
Comment (E-Mac27) [inline internal/common/baseclient/baseSpeaker.go:38]: Please add compulsory / optional to this. If possible, also do the same for all the functions in this file.

File: internal/common/baseclient/baseclient.go [hunk 3]
```diff
@@ -127,7 +128,7 @@ func (c *BaseClient) GetVoteForRule(ruleName string) bool {
 }
 
 // GetVoteForElection returns the client's Borda vote for the role to be elected.
```
Comment (E-Mac27) [inline internal/common/baseclient/baseclient.go:130]: Please add Compulsory / Optional

File: internal/common/baseclient/basejudge.go [hunk 4]
```diff
@@ -99,7 +99,7 @@ func searchForVariableInArray(val rules.VariableFieldName, array []rules.Variabl
 }
 
 // CallPresidentElection is called by the judiciary to decide on power-transfer
```
Comment (E-Mac27) [inline internal/common/baseclient/basejudge.go:83]: Please add compulsory or optional

File: internal/common/baseclient/basepresident.go [hunk 5]
```diff
@@ -53,7 +53,7 @@ func (p *BasePresident) PaySpeaker(salary shared.Resources) (shared.Resources, b
 }
 
 // CallSpeakerElection is called by the executive to decide on power-transfer
```
Comment (E-Mac27) [inline internal/common/baseclient/basepresident.go:59]: Please add compulsory or optional.

File: internal/common/baseclient/baseSpeaker.go [hunk 6]
```diff
@@ -33,12 +33,20 @@ func (s *BaseSpeaker) DecideAnnouncement(ruleId string, result bool) (string, bo
...
 		VotingMethod:  shared.Plurality,
 		IslandsToVote: allIslands,
 		HoldElection:  true,
 	}
+	if monitoring.Performed && monitoring.Result {
+		electionsettings.HoldElection = false
+	}
+	if turnsInPower >= 2 {
+		electionsettings.HoldElection = false
+	}
```
Comment (E-Mac27) [inline internal/common/baseclient/baseSpeaker.go]: contradicting comment? 
electionsettings.HoldElection = true?

Comment (E-Mac27) [reply internal/common/baseclient/baseSpeaker.go]: Same for the other base clients.

**Review by E-Mac27 — CHANGES_REQUESTED** [review]
Just some baseclients things. But overall nothing game breaking.

File: internal/common/baseclient/baseSpeaker.go (hunk 6 above)
Comment (preetl) [reply internal/common/baseclient/baseSpeaker.go]: good spot! yes, this was completely backwards haha. sorted it now!

**Review by E-Mac27 — APPROVED** [review]
Thanks for making the changes

**Comment by lhl2617** [comment]
man, took you over a year to get this done

**Pull request merged**